### PR TITLE
terragrunt: update to 0.52.1

### PR DESCRIPTION
**Pull request by GerardSoleCa**
#### Description

Updating terragrunt to the latest version. It will select terraform 1.5 when installing.

A couple of notes here:

* Current version is broken for me, I cannot end up installing the selected terraform port. That's why I'm upding the port
* Since patch changes, I've done my best to accomodate the fixes. But maybe this is not the standard way to do this in macports. If someone has a better way to do this, please, let me know and I'll update the PR accordingly!
* I've tried running the tests but seems like there are none `Error: Failed to test terragrunt-0.52: terragrunt-0.52 has no tests turned on. see 'test.run' in portfile(7)`
* Although I've configured the version as others, when running `terragrunt -v` says `terragrunt version latest` I suspect that this is due to the release they did

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.0 23A344 arm64
Xcode 15.0 15A240d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: sysutils/terragrunt/Portfile [hunk 1]
```diff
@@ -15,10 +15,23 @@ maintainers         {mjrc.nl:macports @mjrc} \
 # *NOTE* Remember to update `latestVersion` on a version upgrade.
 set latestVersion       terragrunt-0.43
```
Comment (herbygillot) [inline sysutils/terragrunt/Portfile:16]: Looks like you should update this ^

Comment (GerardSoleCa) [reply sysutils/terragrunt/Portfile]: You are right! I've pushed a fix! Thanks for noticing it

**Comment by herbygillot** [comment]
> Since patch changes, I've done my best to accomodate the fixes. But maybe this is not the standard way to do this in macports. If someone has a better way to do this, please, let me know and I'll update the PR accordingly!

Try reverting back to the singular `patchfiles` entry in the body of the Portfile, the way it was before, and then for your `terragrunt-0.52` addition, use `patchfiles-replace`:

```
subport terragrunt-0.52 {
    set dependsOn       1.6
    set patchNumber     1

    patchfiles-replace  patch-options.go.diff patch-options.go.52.diff

    checksums....
}
```

> I've tried running the tests but seems like there are none Error: Failed to test terragrunt-0.52: terragrunt-0.52 has no tests turned on. see 'test.run' in portfile(7)

Tests are failing because this Portfile 1) has no test commands defined, and 2) does not have tests enabled.

You might be able to add testing with something like:

```
build.args          -ldflags \"${go_ldflags}\" -o ${baseName} ${go.package}

test.run            yes
test.cmd            go test
test.args           -v ./...

```

> Although I've configured the version as others, when running terragrunt -v says terragrunt version latest I suspect that this is due to the release they did

The version is embedded into the binary via the ldflags:

```
set go_ldflags      "-X main.VERSION=${patchVersion}"
...
build.args          -ldflags \"${go_ldflags}\" -o ${baseName} ${go.package}
```

If 0.52 changes where the version information is being looked up from, then the version variable being referenced in the ldflag (`main.VERSION`) will need to be changed.

**Comment by GerardSoleCa** [comment]
Hi @herbygillot,

Thanks for the extensive CR. Let me come back with the results and issues I've seen.

**1. patchfiles-replace**

I've tried and it's not working, it tries to patch with `patch-options.go.diff` instead of `patch-options.go.52.diff`. I suspect that it's executed/configured in order of appearence. So, when I setup the replacement on the subport definition, it gets overriden inside the block that configures the whole port.

Logs show that it's using the `patch-options.go.diff`.

I've tried to see if there is a nice way to fix this, but more issues came

**2. Tests**

I confirm that the command you gave works. But after checking the tests and the documentation of terragrunt it might not be a good idea to run the tests. I've seen that tries to connect to AWS and according to the docs it might imply charges. So maybe it's better to keep them disabled.

Apart from these, not everyone might have an aws account.

More info: https://terragrunt.gruntwork.io/docs/community/contributing/#running-tests

**3. Version**

This is similar to the first issue. I've seen how I can configure the version, but it's in a different way than previous ports. I've checked terragrunt commit history and the change happend after the last port we had.

Again, I suspect that the way macports executes the file I cannot set the `build.args` in the subport because it will be overriden by the block that configures the rest.

___________

Do you know if there is a way to configure this differences between subports? I thought in having a variable and then do an if/else. Something like:

```tcl

set newBuildStrategy true

....


else {
    PortGroup           golang 1.0

    supported_archs     x86_64 arm64

    depends_run         port:terragrunt_select port:terraform-${dependsOn}

    set baseVersion     [lindex [split ${subport} "-"] 1]
    set patchVersion    ${baseVersion}.${patchNumber}
    set baseName        terragrunt${baseVersion}

    go.setup            github.com/gruntwork-io/terragrunt ${patchVersion} v

    description         Terragrunt is a thin wrapper for Terraform
    long_description    ${description} that provides extra tools for working \
                        with multiple Terraform modules.

    if (${newBuildStrategy} {
      patchfiles patch-options.go.52.diff
      set go_ldflags      "-X github.com/gruntwork-io/go-commons/version.Version=${patchVersion}"
    } else {
      patchfiles patch-options.go.diff
      set go_ldflags      "-X main.VERSION=${patchVersion}"
    }
}
```

Let's see what do you think about this solution.

**Comment by herbygillot** [comment]
Yes, I think that's fine, let's proceed with that method.

**Comment by herbygillot** [comment]
Also if you would like to keep `terragrunt` up-to-date, you can add yourself as co-maintainer.

**Comment by GerardSoleCa** [comment]
@herbygillot I've pushed the changes we mentioned. It's working :) 

```bash
❯ terragrunt -v
terragrunt version 0.52.1
```

I've also created a new PR with the fixes for terraform 1.2 and 1.3: https://github.com/macports/macports-ports/pull/20834

**Comment by herbygillot** [comment]
Thanks for this submission, @GerardSoleCa.

**Pull request merged**